Dockerfile

```diff
@@ -0,0 +1,8 @@
+FROM python:3.10
+
+WORKDIR /app
+COPY . /app
+
+RUN pip install -r requirements.txt
```

Comment:
I would suggest first copying the requirements in isolation into the /app folder, and running pip install first before copying the rest of the source code.

This would make use of Docker's build caching, which would allow Docker to bypass install dependencies if it notices no changes in the packages, and then copy the rest of the source code, speeding up the build drastically. Check https://stackoverflow.com/questions/61628083/during-build-on-what-basis-does-docker-decide-whether-to-create-a-new-layer-or 

You could also consider utilising Docker's caching support to install packages to cache within the container to further speed up the process.